From: Rene Kita <mail@rkta.de>
Date: Sun Aug 18 09:29:21 CEST 2024
Subject: Add key binding to insert word under cursor
Patch: patches/Add-key-binding-to-insert-word-under-cursor.patch

Press e.g. ^] to insert the word under cursor when inserting text.

---
 linein.c |   10 +++++++++-
 main.c   |    3 ++-
 2 files changed, 11 insertions(+), 2 deletions(-)

--- a/linein.c
+++ b/linein.c
@@ -40,6 +40,7 @@ static void _editor(void);
 static void _enter(void);
 static void _esc(void);
 static void _inbrk(void);
+static void _iword(void);
 static void _mvB(void);
 static void _mvE(void);
 static void _mvL(void);
@@ -73,7 +74,7 @@ void (*InputKeymap[32]) (void) = {
 /*  C-@     C-a     C-b     C-c     C-d     C-e     C-f     C-g     */
     _compl, _mvB,   _mvL,   _inbrk, delC,   _mvE,   _mvR,   _inbrk,
 /*  C-h     C-i     C-j     C-k     C-l     C-m     C-n     C-o     */
-    _bs,    _noop,  _enter, killn,  _noop,  _enter, _next,  _editor,
+    _bs,    _iword, _enter, killn,  _noop,  _enter, _next,  _editor,
 /*  C-p     C-q     C-r     C-s     C-t     C-u     C-v     C-w     */
     _prev,  _quo,   _bsw,   _noop,  _mvLw,  killb,  _quo,   _bsw,
 /*  C-x     C-y     C-z     C-[     C-\     C-]     C-^     C-_     */
@@ -599,6 +600,13 @@ _enter(void)
     i_cont = FALSE;
 }
 
+char * GetWord(Buffer *buf); /* TODO(rkta): Proper declaration from header */
+static void
+_iword(void)
+{
+    ins_char(Strnew_charp(GetWord(Currentbuf)));
+}
+
 static void
 _noop(void)
 {
--- a/main.c
+++ b/main.c
@@ -6018,7 +6018,8 @@ getCurWord(Buffer *buf, int *spos, int *
     return &p[b];
 }
 
-static char *
+char * GetWord(Buffer *buf);
+char *
 GetWord(Buffer *buf)
 {
     int b, e;
